File: NETCore/Cedrus/Controllers/V1/Models/RelationshipTypeModel.cs
using Juniper.Cedrus.Entities;

namespace Juniper.Cedrus.Controllers.V1;

public class RelationshipTypeModel
{
    public string TypeStamp => "relationshipType";
    public int Id { get; }
    public string Name { get; }
    public string ParentRole { get; }
    public string ChildRole { get; }

    public RelationshipTypeModel(RelationshipType? value)
    {
        Id = value?.Id ?? 0;
        Name = value?.Name ?? "Unknown";
        ParentRole = value?.ParentRole ?? "Unknown";
        ChildRole = value?.ChildRole ?? "Unknown"; ;
    }
}

File: JS/cedrus/src/models/RelationshipTypeModel.ts
import { isString } from "@juniper-lib/util";
import { INamed, isINamed } from "./INamed";
import { ITypeStamped, isTypeStamped } from "./ITypeStamped";
import { TypedNameOrId } from "./NameOrId";

export interface RelationshipTypeModel extends INamed, ITypeStamped<"relationshipType"> {
    parentRole: string;
    childRole: string;
}

export function isRelationshipTypeModel(obj: unknown): obj is RelationshipTypeModel {
    return isTypeStamped("relationshipType", obj)
        && isINamed(obj)
        && "parentRole" in obj
        && isString(obj.parentRole)
        && "childRole" in obj
        && isString(obj.childRole);
}

export type RELATIONSHIP_TYPE = TypedNameOrId<"relationshipType">;
export function RELATIONSHIP_TYPE(idOrName: number | string): RELATIONSHIP_TYPE {
    return TypedNameOrId(idOrName, "relationshipType");
}